api-examples/rbacv1/main.py

```diff
@@ -0,0 +1,324 @@
+        """
+        if len(self.validation_errors) != 0:
+            return "found the following validation errors in the request: {}".format(", ".join(self.validation_errors))
+
+        if len(self.message) != 0:
+            return "server responded: {}".format(self.message)
+
+        return "server returned {} code without further information".format(self.code)
+
+
+def handle_status_code(resp=None):
+    """
+    handle_status_code intercepts the response and raises an appropriate error if is not a 200
+
+    :param resp: an http response as returned from requests library
+    :return: None in case of success or raises an exception with details otherwise
+    """
+    if resp is None:
+        return
+    if resp.status_code == 200:
+        return
+    e = SLAPIError(**resp.json())
+    raise Exception(e.as_string())
+
+
+class SLResponse:
+    """
+    Is an implementation of the base 200 response provided by all ShiftLeft API v4 endpoints.
+    """
+
+    def __init__(self, ok=True, response={}):
+        self.ok = ok
+        self.response = response
+
+
+class SLTeamMembership:
+    """
+    SLTeamMembership contains the membership details for a user in a team.
+    """
+
+    def __init__(self, team_name="", team_id="", role="", role_name="", role_aliases=[]) -> None:
+        self.team_name = team_name
+        self.team_id = team_id
+        self.role = role
+        self.role_name = role_name
+        self.role_aliases = role_aliases
+
+
+class SLUser:
+    """
+    SLUser holds the information for one user as returned from ListUsers endpoint in ShiftLeft APIv4
+    https://docs.shiftleft.io/api/#operation/ListOrgRBACUsers
+    """
+
+    def __init__(self, name="", email="", id_v2="", team_membership=[]):
+        self.name = name
+        self.email = email
+        self.id_v2 = id_v2
+        self.team_membership = [SLTeamMembership(**t) for t in team_membership]
+
+    def is_member(self, team=""):
+        for tm in self.team_membership:
+            if tm.team_name == tm:
+                return True
+
+
+class SLListUsersResponse:
+    """
+    SLListUsersResponse represents the response for the ListUsers endpoint in ShiftLeft API v4
+    https://docs.shiftleft.io/api/#operation/ListOrgRBACUsers
+    """
+
+    def __init__(self, users=[]):
+        self.users = [SLUser(**u) for u in users]
+
+    def id_for_email(self, user_email=""):
+        for u in self.users:
+            if u.email.lower() == user_email.lower():
+                return u.id_v2
+
+    def user_for_id(self, user_id=""):
+        for u in self.users:
+            if u.id_v2 == user_id:
+                return u
+
+
+class SLTeamInfo:
+    """
+    SLTeamInfo represents the information returned in one item of ListTeams endpoint in ShiftLeft API v4
+    https://docs.shiftleft.io/api/#operation/ListTeams
+    """
+
+    def __init__(self, team_id="", team_name=""):
+        self.team_id = team_id
+        self.team_name = team_name
```

Comment:
You probably know this, but for these data-only classes you could've done e.g.
```suggestion
SLTeamInfo = namedtuple("SLTeamInfo", ("team_id", "team_name"), defaults=("", ""))
```
in general I think they're a wash for legibility, but for classes of objects that are going to be abundant (e.g. part of a long list) they offer a performance advantage